src/raftstore/store/snap.rs

```diff
@@ -1637,6 +1651,133 @@ mod v2 {
             s2.build(&snapshot, &region, &mut snap_data, &mut stat).unwrap();
             assert!(s2.exists());
         }
+
+        // Make all the snapshot in the specified dir corrupted to have incorrect size.
+        fn corrupt_snapshot_size_in<T: Into<PathBuf>>(dir: T) {
+            let dir_path = dir.into();
+            let read_dir = fs::read_dir(dir_path).unwrap();
+            for p in read_dir {
+                if p.is_ok() {
+                    let e = p.as_ref().unwrap();
+                    if !e.file_name().into_string().unwrap().ends_with(META_FILE_SUFFIX) {
+                        let mut f = OpenOptions::new().append(true).open(e.path()).unwrap();
+                        f.write_all(b"xxxxx").unwrap();
+                    }
+                }
+            }
+        }
+
+        // Make all the snapshot in the specified dir corrupted to have incorrect checksum.
+        fn corrupt_snapshot_checksum_in<T: Into<PathBuf>>(dir: T) -> Vec<SnapshotMeta> {
+            let dir_path = dir.into();
+            let mut res = Vec::new();
+            let read_dir = fs::read_dir(dir_path).unwrap();
+            for p in read_dir {
+                if p.is_ok() {
+                    let e = p.as_ref().unwrap();
+                    if e.file_name().into_string().unwrap().ends_with(META_FILE_SUFFIX) {
+                        let mut snapshot_meta = SnapshotMeta::new();
```

Comment:
The corruption could be arbitrary. It take a lot of effort to cover all of them.
The test case added is to simulate the small corruption of size/checksum, which relates to this change in this PR.